src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -60,7 +60,20 @@ public function execute()
 
             $con = $this->connection->getResource();
 
-            $this->search = @ldap_search(
+            switch ($this->options['scope']) {
+                case static::SCOPE_BASE:
+                    $func = 'ldap_read';
+                    break;
+                case static::SCOPE_ONELEVEL:
+                    $func = 'ldap_list';
+                    break;
+                case static::SCOPE_SUBTREE:
+                default:
```

Comment:
Actually I already defined allowed values in option OptionsResolver (https://github.com/symfony/symfony/pull/20310/commits/c6c47aec4f2520a2d54ca38391dcc7c33f703bcf#diff-822ec15d626e08f71c9ebaea383b5a89R40). So default is only for linter.